crates/subspace-networking/src/protocols/connected_peers.rs

```diff
@@ -229,13 +293,33 @@ impl<Instance> Behaviour<Instance> {
     /// Specifies the whether we should keep connections to the peer alive. The decision could
     /// depend on another protocol (e.g.: PeerInfo protocol event handling).
     pub fn update_keep_alive_status(&mut self, peer_id: PeerId, keep_alive: bool) {
-        let not_enough_connected_peers =
-            self.permanently_connected_peers() < self.config.max_connected_peers;
```

Comment:
This should have simply changed to something like this:
```suggestion
        let not_enough_connected_peers =
            self.permanently_connected_peers() < self.config.target_connected_peers ||
            (
                connection_state.connection_type() == ConnectionType::Incoming && self.permanently_connected_peers() < self.config.max_connected_peers
            );
```

Reply:
It's less obvious IMHO but no problem. I'm adding an additional comment.

Reply:
Different preferences. I use another logic sequence for the same result. This is why I added a comment for my future self.